jdaviz/configs/imviz/wcs_utils.py

```diff
@@ -275,3 +286,180 @@ def data_outside_gwcs_bounding_box(data, x, y):
         if not (bb_xmin <= x <= bb_xmax and bb_ymin <= y <= bb_ymax):
             outside_bounding_box = True  # Has to be Python bool, not Numpy bool_
     return outside_bounding_box
+
+
+def _get_fits_wcs_from_file(filename):
+    header = fits.getheader(filename)
+    with warnings.catch_warnings():
+        # Ignore a warning on using DATE-OBS in place of MJD-OBS
+        warnings.filterwarnings('ignore', message="'datfix' made the change",
+                                category=FITSFixedWarning)
+        wcs = FitsWCS(header)
+
+    return wcs
+
+
+def rotated_gwcs(
+    center_world_coord,
+    rotation_angle,
+    pixel_scales,
+    cdelt_signs,
+    refdata_shape=(2, 2)
+):
+    # based on ``gwcs_simple_imaging_units`` in gwcs:
+    #   https://github.com/spacetelescope/gwcs/blob/
+    #   eec9a2b6de8356495f405de3dc6531538589ce5d/gwcs/tests/conftest.py#L165
+    rho = rotation_angle
+    sin_rho = np.sin(rho.to_value(u.rad))
+    cos_rho = np.cos(rho.to_value(u.rad))
+    rotation_matrix = np.array([[cos_rho, -sin_rho],
+                                [sin_rho, cos_rho]])
+
+    # "rescale" the pixel scales. Scaling constant was tuned so that the
+    # synthetic image is about the same size on the sky as the input image
+    rescale_pixel_scale = np.array(refdata_shape) / 1000

```

Comment:
1000 seems too magical a number. How was this tuned exactly and when should we update this number in the future?